internal/configurations/configurations.go

```diff
@@ -235,12 +235,15 @@ func (c *Configuration) GetSecret(ctx context.Context) (*v1.Secret, error) {
 	return secret, nil
 }
 
-// ForService returns a slice of Secrets matching the given Service.
+// ForService returns a slice of configuration secrets matching the given Service.
 func ForService(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service) ([]v1.Secret, error) {
 	secretSelector := labels.Set(map[string]string{
-		"app.kubernetes.io/instance": names.ServiceHelmChartName(service.Meta.Name, service.Meta.Namespace),
+		"app.kubernetes.io/instance": names.ServiceReleaseName(service.Meta.Name),
```

Comment:
I haven't tried but the `in` operator ([set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement)) should help us avoiding two requests and merging the values:

```go
secretSelector := labels.Set(map[string]string{
	ConfigurationLabelKey:        "true",
	ConfigurationTypeLabelKey:    "service",
}).AsSelector()
	
secretSelector.Add(
	labels.NewRequirement(
		"app.kubernetes.io/instance", 
		selection.In, 
		[]string{
			names.ServiceReleaseName(service.Meta.Name),
			names.ServiceHelmChartName(service.Meta.Name, service.Meta.Namespace)
		},
	),
)
```

Reply:
Nice. Will check.

Reply:
Done. Commit 520c57020976e9267154fb21ff968e37371155ad. Let us see what CI thinks of this.

Reply:
This has been undone. See commit c1e0a61b4da503bd427fc250310958dc7c89a7f8 for in-code explanations.

Reply:
Having again a look at the code the problem was in my wrong suggestion, that was assuming that the `secretSelector.Add(*multiLabel)` was going to change the `secretSelector`, while instead is returning a new Selector.

Also I've seen that the two functions are basically the same, so we could simplify them using a common `forService` func:

```go
// ForService returns a slice of configuration secrets matching the given Service.
func ForService(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service) ([]v1.Secret, error) {
	// The difference to `ForServiceUnlabeled` is here. We are explicitly looking for
	// the labels attached to the secrets by `LabelServiceSecrets` as we want only
	// proper configurations.
	secretSelector := labels.Set(map[string]string{
		ConfigurationLabelKey:     "true",
		ConfigurationTypeLabelKey: "service",
	}).AsSelector()

	return forService(ctx, kubeClient, service, secretSelector)
}

// ForServiceUnlabeled returns a slice of unlabeled secrets matching the given Service
func ForServiceUnlabeled(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service) ([]v1.Secret, error) {
	// The difference to `ForService` is here. Not looking for the labels attached to
	// the secrets by `LabelServiceSecrets` to turn them into configurations.
	return forService(ctx, kubeClient, service, labels.NewSelector())
}

func forService(ctx context.Context, kubeClient *kubernetes.Cluster, service *models.Service, secretSelector labels.Selector) ([]v1.Secret, error) {

	// Note how we search here for both regular and old-style secrets in one set-based selector
	// and call to come. No need to perform two requests and merge the results.
	//
	// COMPATIBILITY SUPPORT for services from before https://github.com/epinio/epinio/issues/1704 fix
	// Look for secrets referencing a (helm controller)-based service.

	multiLabel, err := labels.NewRequirement(
		"app.kubernetes.io/instance",
		selection.In,
		[]string{
			names.ServiceReleaseName(service.Meta.Name),
			names.ServiceHelmChartName(service.Meta.Name, service.Meta.Namespace),
		},
	)
	if err != nil {
		return nil, err
	}

	secretSelector = secretSelector.Add(*multiLabel)

	listOptions := metav1.ListOptions{
		LabelSelector: secretSelector.String(),
	}

	secretList, err := kubeClient.Kubectl.CoreV1().Secrets(service.Meta.Namespace).List(ctx, listOptions)
	if err != nil {
		return nil, err
	}

	filteredSecrets := filterSecretsByType(secretList.Items, service.SecretTypes)

	return filteredSecrets, nil
}
```

Reply:
I see. Ok. Will trial that.

Reply:
> while instead is returning a new Selector.

And the selector going in was the empty selector, matching everything :facepalm:

Reply:
New trial pushed. Commit be612559d43f318224e4bd052b49e4b5ea549530